src/ninja.cc

```diff
@@ -49,20 +49,20 @@
 
 static size_t FindCommandLineFlag(StringPiece cmd, StringPiece name) {
   const size_t found = cmd.find(name);
-  if (found == string::npos || found == 0)
-    return string::npos;
+  if (found == std::string::npos || found == 0)
```

Comment:
How about `using std::string` and the like at the beginning of compilation units? It would make the code a bit more concise while not being subject to the disadvantages of `using namespace std;`

Reply:
I am personally very much against shortcutting the std:: namespace as it often gives no obvious benefit, but makes the code ambiguous. If type names get lengthy and unreadable and are reused often, they should anyway be captured in aliased types.

The Google style guide says: 
_You may not use a using-directive to make all names from a namespace available._
https://google.github.io/styleguide/cppguide.html#Namespaces

Now, we could do `using std::string; using std::vector;` etc, but unless you insist I would rather not go down this road.

Reply:
FWIW, using `std::vector` is what seems to be the standard at Google, so it's probably not that bad.

That said, this is not google3 code, so all that matters is that it's consistent with itself.